server/src/service.rs

```diff
@@ -115,17 +116,37 @@ impl Service {
     #[zbus(out_args("unlocked", "prompt"))]
     pub async fn unlock(
         &mut self,
-        _objects: Vec<OwnedObjectPath>,
-    ) -> Result<(Vec<OwnedObjectPath>, ObjectPath), ServiceError> {
-        todo!()
+        objects: Vec<OwnedObjectPath>,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+    ) -> Result<(Vec<OwnedObjectPath>, OwnedObjectPath), ServiceError> {
+        let unlocked = self.set_locked(false, &objects).await?;
+        if unlocked.is_empty() {
+            let prompt = Prompt::new(Arc::clone(&self.manager)).await;
+            let path = prompt.path().clone();
+            object_server.at(path.clone(), prompt).await?;
+
+            return Ok((unlocked, path));
+        }
+
+        Ok((unlocked, OwnedObjectPath::default()))
     }
 
     #[zbus(out_args("locked", "prompt"))]
     pub async fn lock(
         &mut self,
-        _objects: Vec<OwnedObjectPath>,
-    ) -> Result<(Vec<OwnedObjectPath>, ObjectPath), ServiceError> {
-        todo!()
+        objects: Vec<OwnedObjectPath>,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+    ) -> Result<(Vec<OwnedObjectPath>, OwnedObjectPath), ServiceError> {
+        let locked = self.set_locked(true, &objects).await?;
+        if locked.is_empty() {
+            let prompt = Prompt::new(Arc::clone(&self.manager)).await;
+            let path = prompt.path().clone();
+            object_server.at(path.clone(), prompt).await?;
+
+            return Ok((locked, path));
+        }
```

Comment:
Tbh, all the pormpt related bits should be removed from this commit and included in a separate patchset that handles all the prompt related parts. Currently, you stopped locking/unlocking the items and left unused code while adding a non-working functionality

Reply:
Done.